docs/spec/openapi.yaml

```diff
@@ -322,6 +322,42 @@ components:
             source document, or a specific snippet.
       required:
         - kind
+    TrajectoryMetadata:
+      type: object
+      description: >
+        Represents trajectory information for an agent's reasoning or tool execution
+        steps. This metadata helps track the agent's decision-making process and
+        provides transparency into how the agent arrived at its response.
+
+        TrajectoryMetadata can capture either:
+        1. A reasoning step with a message
+        2. A tool execution with tool name, input, and output
+
+        This information can be used for debugging, audit trails, and providing
+        users with insight into the agent's thought process.
+      properties:
+        kind:
+          type: string
+          const: trajectory
+          default: trajectory
+        message:
+          type: string
+          nullable: true
+          description: A reasoning step or thought in the agent's decision process.
```

Comment:
```suggestion
          description: Trajectory message; For example reasoning step or thought in the agent's decision process.
```

Reply:
Similar observation to above, I'd avoid "decision process" and replace with a term that's more generic.  I suggested "operational information" above as that can encompass decisions, tools, and a vast variety of other things the agent might wish to log.